src/DevouchAttester.sol

```diff
@@ -1,79 +1,105 @@
     }
 
     function updateFee(uint256 newFee) public ownerOnly {
         fee = newFee;
-        emit Log("updateFee", gasleft());
     }
 
-    function updateSchema(bytes32 newSchema) public ownerOnly{
+    function updateSchema(bytes32 newSchema) public ownerOnly {
+        if (newSchema == 0) revert InvalidSchema();
         schema = newSchema;
-        emit Log("updateSchema", gasleft());
     }
 
-    function attest(address recipient, bytes32 refUID, bytes calldata data) public payable
-    {
-        require(address(msg.sender).balance >= fee, "Insufficient balance to pay fee");
-        require(msg.value == fee, "Must pay the fee amount");
-
-        IEAS(0xC2679fBD37d54388Ce493F1DB75320D236e1815e).attest(
-                AttestationRequest({
-                    schema: schema,
-                    data: AttestationRequestData({
-                        recipient: recipient,
-                        expirationTime: NO_EXPIRATION_TIME,
-                        revocable: true,
-                        refUID: refUID,
-                        data: data,
-                        value:0
-                    })
-                })
-            );
-        
-        emit Log("attested", gasleft());
+    function updateEAS(address newEASAddress) public ownerOnly {
+        eas = IEAS(newEASAddress);
     }
 
-    receive() external payable {}
+    function togglePause() public ownerOnly {
+        paused = !paused;
+    }
 
-    fallback() external payable {
-        emit Log("fallback", gasleft());
+    function attest(
+        address recipient,
+        bytes32[] calldata refUIDArray,
+        bytes calldata data
+    ) public payable whenNotPaused {
+        if (msg.value != fee) revert InvalidFee();
+
+        uint256 len = refUIDArray.length;
+        AttestationRequestData[] memory requestDataArray = new AttestationRequestData[](len);
+
+        for (uint256 i = 0; i < len; ++i) {
+            requestDataArray[i] = AttestationRequestData({
+                recipient: recipient,
+                expirationTime: NO_EXPIRATION_TIME,
+                revocable: true,
+                refUID: refUIDArray[i],
+                data: data,
+                value: 0
+            });
+        }
+
+        MultiAttestationRequest[] memory multiRequests = new MultiAttestationRequest[](1);
+        multiRequests[0] = MultiAttestationRequest({
+            schema: schema,
+            data: requestDataArray
+        });
+
+        eas.multiAttest(multiRequests);
     }
 
+    receive() external payable {}
+
+    fallback() external payable {}
+
     // Function to withdraw Ether from the contract (for testing purposes)
-    function withdraw() public {
-        // Transfer the Ether to the multisig address
-        (bool success, ) = multisig.call{value: address(this).balance}("");
-        require(success, "Failed to send Ether");
-        emit Log("withdraw", gasleft());
+    function withdraw() public ownerOnly {
```

Comment:
@jainkrati @mateodaza is there also a ownerOnly function to change the target recipient then? In case we need to change which multisig receives the funds. Apologies if it's already in the code somewhere

Reply:
nope, good catch - will add it 👍🏼